src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java

```diff
@@ -0,0 +1,132 @@
+      }
+      try {
+        Optional<Bundle.BundleEntryComponent> measureEntry = getMeasureEntry(bundleResource);
+        if (measureEntry.isEmpty()) {
+          log.error("Unable to find measure entry for measure {}", madieMeasure.getId());
+          throw new ResourceNotFoundException("measure entry", madieMeasure.getId());
+        }
+        Resource measureResource = measureEntry.get().getResource();
+
+        // converting measure resource from R4 to R5
+        var versionConvertor_40_50 = new VersionConvertor_40_50(new BaseAdvisor_40_50());
+        org.hl7.fhir.r5.model.Measure r5Measure =
+            (org.hl7.fhir.r5.model.Measure) versionConvertor_40_50.convertResource(measureResource);
+
+        String effectiveDataRequirementsStr =
+            elmTranslatorClient.getEffectiveDataRequirements(
+                bundleResource,
+                madieMeasure.getCqlLibraryName(),
+                accessToken,
+                madieMeasure.getId());
+
+        org.hl7.fhir.r5.model.Library effectiveDataRequirements =
+            fhirContextForR5
+                .newJsonParser()
+                .parseResource(org.hl7.fhir.r5.model.Library.class, effectiveDataRequirementsStr);
+
+        // effectiveDataRequirements needs to have fixed id: effective-data-requirements
+        effectiveDataRequirements.setId("effective-data-requirements");
+        r5Measure.addContained(effectiveDataRequirements);
+        r5Measure.getExtension().add(createExtension());
+
+        String measureTemplate = getData("/templates/Measure.liquid");
+        LiquidEngine engine = getLiquidEngine(madieMeasure);
+        LiquidEngine.LiquidDocument doc = engine.parse(measureTemplate, "hr-script");
+        String measureHr = engine.evaluate(doc, r5Measure, null);
+        // Wrapper template for Measure.liquid o/p
+        String humanReadable = getData("/templates/HumanReadable.liquid");
+        return humanReadable.replace("human_readable_content_holder", measureHr);
+      } catch (FHIRException fhirException) {
+        log.error(
+            "Unable to generate Human readable for measure {} Reason => {}",
+            madieMeasure.getId(),
+            fhirException);
+        throw new HumanReadableGenerationException("measure", madieMeasure.getId());
+      }
+    } else {
+      log.error("Unable to find a bundleResource for measure {}", madieMeasure.getId());
+      throw new ResourceNotFoundException("bundle", madieMeasure.getId());
+    }
+  }
+
+  /**
+   * @param bundleResource Bundle resource
+   * @return BundleEntry which is of type Measure
+   */
+  private Optional<Bundle.BundleEntryComponent> getMeasureEntry(Bundle bundleResource) {
+    return bundleResource.getEntry().stream()
+        .filter(
+            entry ->
+                StringUtils.equalsIgnoreCase(
+                    "Measure", entry.getResource().getResourceType().toString()))
+        .findFirst();
+  }
+
+  private Extension createExtension() {
```

Comment:
```suggestion
  private Extension createEffectiveDataRequirementExtension() {
```
More detailed name to avoid accidentally using it in some other HR section.